core/Db/TransactionLevel.php

```diff
@@ -40,6 +40,10 @@ public function canLikelySetTransactionLevel()
 
     public function setUncommitted()
     {
+        if ($this->db->supportsUncommitted !== null) {
+            return $this->db->supportsUncommitted;
```

Comment:
If we also "cache" and return if `supportsUncommitted` is `true`, that means we won't perform the setting of the transaction level anymore. That might be fine, but don't we need to unset `supportsUncommitted` then in `restorePreviousStatus`, to ensure it's checked and set again if needed?

Reply:
good point, we still want to set the transaction mode, but we don't need to execute the `Option::set` for example to check if it worked.

Reply:
@tsteur @sgiehl did a little adjustment, would that work